src/client/components/localisation/confidence_ellipse/model.ts

```diff
@@ -0,0 +1,46 @@
+import { observable } from 'mobx'
+import { Vector3 } from '../../../math/vector3'
+
+export class ConfidenceEllipseModel {
+  @observable public color: string
+  @observable public opacity: number
+  @observable public position: Vector3
+  @observable public scaleX: number
+  @observable public scaleY: number
+  @observable public rotationAxis: Vector3
+  @observable public rotationAngle: number
+
+  public constructor(opts: Partial<ConfidenceEllipseModel>) {
+    Object.assign(this, opts)
+  }
+
+  public static of({
+    color = '#0000ff',
+    opacity = 0.5,
+    position = new Vector3(0, 0, 0.001123),
```

Comment:
🚗 🔫  Can probably move the offset concern to the view model by using `polygonOffset` on the material.

Example: http://jsfiddle.net/adt1d5b7/1/

Reference: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glPolygonOffset.xhtml

Disclaimer: I haven't used it myself, but I have been meaning to solve this issue, and will probably use it for the field lines.